armi/reactor/components/component.py

```diff
@@ -1019,8 +1031,20 @@ def mergeNuclidesInto(self, compToMergeWith):
             )
 
     def iterComponents(self, typeSpec=None, exact=False):
-        if self.hasFlags(typeSpec, exact):
-            yield self
+        """
+        Yield the component or its direct child components.
+
+        Notes
+        -----
+        This could probably be made recursive to allow arbitrary depth, but it's more complex
+        """
+        if self._children:
+            for child in self._children:
+                if self.hasFlags(typeSpec, exact):
+                    yield child
+        else:
+            if self.hasFlags(typeSpec, exact):
+                yield self
```

Comment:
@ntouran we've found that the mass of the matrix material is not accounted for when doing `getMass` calls. I think due to this section here

We modified the triso mass test blueprint file such that each element only appears in one `Component`. Then, `block.getMass("C12")` (in theory) picks up mass from the `triso compact` component and should be non-zero. Unfortunately this is not the case 😬 

```yaml
# A full blueprint file that contains assems made with ComponentGroups
blocks:
    fuel2: &block_fuel2
        triso compact: 
            shape: Circle
            blends: 
              - triso particle
            blendFracs: 
              - 0.70
            material: Graphite
            Tinput: 600.0
            Thot: 600.0
            mult: 1
            id: 0.0
            od: 1.2446
        duct: 
            shape: Hexagon
            material: Cu
            Tinput: 600.0
            Thot: 600.0
            ip: 9.0
            mult: 1.0
            op: 10.0
        coolant: 
            shape: DerivedShape
            material: NaCl
            Tinput: 25.0
            Thot: 600.0

components:
    # triso dims from Table 4 in TRISO benchmark
    kernel:
        shape: Sphere
        material: Lead
        Tinput: 600.0
        Thot: 600.0
        id: 0.0
        mult: 1.0
        od: 0.010625
    buffer:
        shape: Sphere
        material: Lead
        Tinput: 600.0
        Thot: 600.0
        id: kernel.od
        mult: 1.0
        od: 0.015625
    ipyc:
        shape: Sphere
        material: Lead
        Tinput: 600.0
        Thot: 600.0
        id: buffer.od
        mult: 1.0
        od: 0.017375
    sic:
        shape: Sphere
        material: Lead
        Tinput: 600.0
        Thot: 600.0
        id: ipyc.od
        mult: 1.0
        od: 0.019125
    opyc:
        shape: Sphere
        material: Lead
        Tinput: 600.0
        Thot: 600.0
        id: sic.od
        mult: 1.0
        od: 0.021125

component groups:
    triso particle:
      kernel:
        mult: 1
      buffer:
        mult: 1
      ipyc:
        mult: 1
      sic:
        mult: 1
      opyc:
        mult: 1


assemblies:
    fuel c: &assembly_c
        specifier: OC
        blocks: [*block_fuel2]
        height: [1.0]
        axial mesh points: [1]
        xs types: [A]
systems:
    core:
        grid name: core
        origin:
            x: 0.0
            y: 0.0
            z: 0.0
grids:
    core:
        geom: hex
        symmetry: full
        lattice map: |
          OC


nuclide flags:
    B: &nuc_flags {burn: false, xs: true}
    C: *nuc_flags
    PB: *nuc_flags
    CU63: *nuc_flags
    CU65: *nuc_flags
    CL35: *nuc_flags
    CL37: *nuc_flags
    CL37: *nuc_flags
    NA23: *nuc_flags
```

Reply:
Thanks @drewj-usnctech I will bring in your modified blueprints file and update the unit test so it fails as you've found and then try to fix.